docs/docs/components-and-props.md

```diff
@@ -0,0 +1,247 @@
+---
+id: components-and-props
+title: Components and Props
+permalink: docs/components-and-props.html
+prev: rendering-elements.html
+---
+
+Components let you split the UI into independent reusable pieces, and think about each piece separately as if it existed in the isolation.
```

Comment:
There should be a comma between the two modifiers of "pieces": "independent, reusable pieces.

That last part should be "as if it existed in isolation" without "the" but I wonder if it's redundant. How about "...and think about each piece in isolation."